### PR TITLE
#859 Remove spacer, closes #859

### LINKED ISSUE #859: Remove $spacer to simpler rem values
### Problem

There is incistency with value sometime using $spacer sometime rem and the actual solution like $spacer*1.5 or $spacer-md - $spacer-xs or $spacer-mdxs are far from ideal.

### Solution

Replace all $spacer-md by 1rem etc... which is much simpler

### PR DESCRIPTION
**Pull request by pieer**

### CODE REVIEW

**Comment by sandrina-p** [comment]
@pieer this PR has 80 commits... It seems you created this branch from your previous PR branch (#851) instead of master. I think this will create massive conflicts when we try to merge it with master.

To simplify you can just create a new brand branch (from master!) and cherry pick this single commit "75595de - Remove all spacer variable except form" to it and create a new PR.

File: frontend/assets/style/_variables.scss [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// TODO remove once Sandrina changes are done (next commit)
```
Comment (sandrina-p) [inline frontend/assets/style/_variables.scss]: 👍

**Comment by pieer** [comment]
It's already merge with master @sandrina-p. And if you just squash merge you won't see any difference I believe?

File: frontend/assets/style/components/_cards.scss [hunk 2]
```diff
@@ -19,7 +19,37 @@
...
 .card-header {
-  margin-bottom: $spacer-sm;
+  margin-bottom: 0.5rem;
+}
+
+@mixin removeSideCardPadding {
+  width: calc(100% + 2rem);
+  margin-left: -0.5rem;
+  margin-right: -0.5rem;
+
```
Comment (sandrina-p) [inline frontend/assets/style/components/_cards.scss]: This code is new... but I think it was already in master. What's wrong here?

File: frontend/views/components/Page.vue [hunk 3]
```diff
@@ -120,13 +120,13 @@ $pagePaddingDesktop: 5.5rem;
   }
 
   @include tablet {
-    width: calc(100% - #{$spacer-xl});
+    width: calc(100% - #{4rem});
```
Comment (sandrina-p) [inline frontend/views/components/Page.vue]: no need to have `#{}` here

File: frontend/views/components/menu/MenuHeader.vue [hunk 4]
```diff
@@ -13,7 +13,7 @@ export default {
 @import "@assets/style/_variables.scss";
 
 .c-header {
-  padding: $spacer-sm $spacer $spacer-sm $spacer;
+  padding: 0.5rem 1rem 0.5rem 1rem;
```
Comment (sandrina-p) [inline frontend/views/components/menu/MenuHeader.vue]: `padding: 0.5rem 1rem` is enough

File: frontend/views/pages/DesignSystem.vue [hunk 5]
```diff
@@ -459,15 +443,15 @@ page(
         tr
           td
             pre
-              | $spacer-lg
+              | 2rem
```
Comment (sandrina-p) [inline frontend/views/pages/DesignSystem.vue]: Odd... i think this line doesn't make sense and it should be removed.

**Review by sandrina-p — CHANGES_REQUESTED** [review]
What I've seen it's all good, it seems a pretty direct find-replace approach.
Still, I noticed `$spacer` variables around still existing. Look for `$spacer` in the whole project. There are still 24 files to change.

image

**Comment by taoeffect** [comment]
Wait, what issue is this closing? It's important that the problem/solution be clearly documented.

**Comment by sandrina-p** [comment]
@taoeffect this PR closes https://github.com/okTurtles/group-income-simple/issues/859

File: frontend/assets/style/components/_cards.scss (hunk 2 above)
Comment (pieer) [reply frontend/assets/style/components/_cards.scss]: Nothing, all good

**Review by sandrina-p — DISMISSED** [review]
Yey, one step closer to say bye-bye to `$spacer` variables :D

**Review by taoeffect — APPROVED** [review]
Fantastic! 💪

**Pull request merged**